processor/resourcedetectionprocessor/resourcedetection_processor.go

```diff
@@ -23,7 +28,17 @@ type resourceDetectionProcessor struct {
 	schemaURL          string
 	override           bool
 	httpClientSettings confighttp.ClientConfig
+	refreshInterval    time.Duration
 	telemetrySettings  component.TelemetrySettings
+
+	stopCh  chan struct{}
+	wg      sync.WaitGroup
+	current atomic.Value
```

Comment:
Consider using atomic.Pointer, since you are storing a pointer. It also doesn't have the gotcha about writing after calling store.